src/Tests/TestProfiler/ClassFactory.cpp

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "ClassFactory.h"
+#include "EnvProfiler/EnvProfiler.h"
+#include "ProfilerBase.h"
+
+ClassFactory::ClassFactory(REFCLSID guidClsid) :
```

Comment:
Do you expect the test profiler to be very different from the real one? Perhaps we can make the real one as well (or first).

Reply:
I think we can reuse the `ProfilerBase` "base" class in product too. Maybe the `ClassFactory` as well if we want to `#ifdef` between test and product or come up with some registration mechanism.